wewatch-client/src/views/swiping/swiping-container.js

```diff
@@ -50,124 +54,162 @@ const SwipingContainer = (props) => {
     setmediaType(topTenMovies[index].type);
     setDescription(topTenMovies[index].synopsis);
   });
-  
+
   const onClickDislike = () => {
-    socket.emit('dislike_event', {roomId: roomId});
+    socket.emit('dislike_event', { roomId: roomId });
     iterateMovie(index);
   };
 
   const onClickLike = () => {
     const movieId = topTenMovies[index].netflixid;
-    socket.emit('like_event', {roomId: roomId, movieId: movieId, movieData: topTenMovies[index]});
+    socket.emit('like_event', { roomId: roomId, movieId: movieId, movieData: topTenMovies[index] });
     iterateMovie(index);
   }
 
   //listen to the no match event
   useEffect(() => {
     socket.on('noMatchRedirect', () => {
-        setNoMatch(true);
-   });
-   return () => { socket.off('noMatchRedirect')};
+      setNoMatch(true);
+    });
+    return () => { socket.off('noMatchRedirect') };
+  });
+
+  useEffect(() => {
+    //listen to match event
+    socket.on('matchRedirect', ({ matchedMovieId, matchedMovieData }) => {
+      setMatchedMovie(matchedMovieData);
+      setMatch(true);
+    });
+
+    return () => { socket.off('matchRedirect') };
+  })
+
+  const endSessionEmit = () => {
+    socket.emit('try_again_event', { roomId: roomId })
+  }
+
+  //listen to try again event
```

Comment:
Can potentially remove this comment since the 'socket.on' function makes it clear that the component would be listening for a socket signal, as well as which signal if would be listening to. I think this helps keep our code clean, what are your thoughts?